contracts/upgradeable_contracts/BasicHomeBridge.sol

```diff
@@ -74,9 +77,24 @@ contract BasicHomeBridge is EternalStorage, Validatable {
         if (signed >= reqSigs) {
             setNumMessagesSigned(hashMsg, markAsProcessed(signed));
             emit CollectedSignatures(msg.sender, hashMsg, reqSigs);
+
+            address feeManager = feeManagerContract();
+            if (feeManager != address(0)) {
+                handleSignatureFeeDistribution(feeManager, message);
+            }
         }
     }
 
+    function handleSignatureFeeDistribution(address feeManager, bytes message) internal {
+        address recipient;
+        uint256 amount;
+        bytes32 txHash;
+        address contractAddress;
+        (recipient, amount, txHash, contractAddress) = Message.parseMessage(message);
+        uint256 fee = calculateFee(amount, true, feeManager);
+        require(feeManager.delegatecall(abi.encodeWithSignature("distributeFeeFromSignatures(uint256)", fee)));
```

Comment:
Consider to have `distributeFeeFromSignatures` and `distributeFeeFromAffirmation` similarly you implemented `calculateFee` below in this contract. It will make the code more readable.

Reply:
Done as part of this commit https://github.com/poanetwork/poa-bridge-contracts/pull/132/commits/faf9712db27d7f3a22eda384f2131368791dbfe1